server/controllers/auth.js

```diff
@@ -121,41 +122,75 @@ exports.loadUser = asyncHandler(async (req, res, next) => {
 exports.logoutUser = asyncHandler(async (req, res, next) => {
   res.clearCookie("token");
 
-  res.send("You have successfully logged out");
+  res.status(200).json({
+    success: {
+      message: "You have successfully logged out",
+    },
+  });
 });
 
 // @route GET /auth/demo-user
 // @desc Load demo user
 // @access Private
 exports.loadDemoUser = asyncHandler(async (req, res, next) => {
-  let demoUser
-  const demoUserExists = await User.findOne({email: "demoUser@email.com"})
+  let demoUser;
+  const demoUserExists = await User.findOne({ email: "demoUser@email.com" });
 
   if (!demoUserExists) {
     demoUser = await User.create({
       email: "demoUser@email.com",
       username: "demoUser",
-      password: "demoUser123"
-    })
+      password: "demoUser123",
+    });
   } else {
-    demoUser = demoUserExists
+    demoUser = demoUserExists;
   }
 
   const token = generateToken(demoUser._id);
   const secondsInWeek = 604800;
 
   res.cookie("token", token, {
     httpOnly: true,
-    maxAge: secondsInWeek * 1000
+    maxAge: secondsInWeek * 1000,
   });
 
   res.status(200).json({
     success: {
       user: {
         id: demoUser._id,
         username: demoUser.username,
-        email: demoUser.email
-      }
-    }
+        email: demoUser.email,
+      },
+    },
+  });
+});
+
+// @route PATCH /auth/update
+// @desc update a user's password
+// @access Private
+exports.updatePassword = asyncHandler(async (req, res, next) => {
+  const { password, oldPassword } = req.body;
+
+  const user = await User.findById(req.user.id).select("+password");
+
+  if (!user) {
+    res.status(400);
+    throw new Error("The user with this ID doesn't exist");
+  }
+
+  if (!(await user.isMatch(oldPassword, user.password))) {
+    res.status(400);
+    throw new Error("Incorrect password");
+  }
```

Comment:
This is cool. But I think a `401` will be more suitable for this status since the request is in order, but incorrect information was provided.

Reply:
@ugur-akin I think 403 is a better response since the client's identity is known they are only unauthorized.